src/sentry/coreapi.py

```diff
@@ -148,15 +149,17 @@ def project_id_from_auth(self, auth):
         return self.project_key_from_auth(auth).project_id
 
     def ensure_does_not_have_ip(self, data):
-        if 'request' in data:
-            if 'env' in data['request']:
-                data['request']['env'].pop('REMOTE_ADDR', None)
+        env = get_path(data, 'request', 'env')
+        if env:
+            env.pop('REMOTE_ADDR', None)
```

Comment:
(here and below)
Can`env` theoretically not be a dictionary here? E.g. when the event is malformed.

Reply:
It can be either `None` or a dict.